app/routes/reports.tsx

```diff
@@ -92,21 +197,85 @@ export default function Reports() {
 
 			<ReportsHeader reports={reports} amounts={contributionAmounts.amounts} />
 
-			<section className="grid grid-rows-1 md:grid-cols-2 xl:grid-cols-3 2xl:grid-cols-4 gap-5 md:gap-3 max-w-screen-xl">
-				{reports.map((report: Report) => (
-					<Link to={`/reports/${report.slug}`} key={report.hypercertId}>
-						<ReportCard
-							hypercertId={report.hypercertId}
-							image={report.image}
-							title={report.title}
-							summary={report.summary}
-							category={report.category}
-							state={report.state}
-							totalCost={report.totalCost}
-							fundedSoFar={report.fundedSoFar}
+			<section className="pb-16 md:pb-8">
+				<div className="grid grid-rows-1 md:grid-cols-2 xl:grid-cols-3 2xl:grid-cols-4 gap-5 md:gap-3 max-w-screen-xl">
+					{getSelectedReports.length
+						? pageTransactions.map((report: Report) => (
+								<Link
+									to={`/reports/${report.slug}`}
+									key={report.hypercertId}
+									prefetch="intent"
+								>
+									<ReportCard
+										hypercertId={report.hypercertId}
+										image={report.image}
+										title={report.title}
+										summary={report.summary}
+										category={report.category}
+										state={report.state}
+										totalCost={report.totalCost}
+										fundedSoFar={report.fundedSoFar}
+									/>
+								</Link>
+						  ))
+						: null}
+				</div>
+
+				{/* Pagination */}
+				{needsPagination && (
+					<Pagination className="pt-6">
+						<PaginationContent>
+							<PaginationItem className="hover:cursor-pointer">
+								<PaginationPrevious
+									onClick={() =>
+										currentPage > 1 ? loadPage(currentPage - 1) : null
+									}
+								/>
+							</PaginationItem>
+							{pageNumbers.map((pageNum, index) => (
+								<PaginationItem
+									onClick={() => loadPage(pageNum)}
+									className="hover:cursor-pointer"
+									key={`page-${pageNum}`}
+								>
+									<PaginationLink isActive={currentPage === pageNum}>
+										{pageNum}
+									</PaginationLink>
+								</PaginationItem>
+							))}
+							{maxPage > 4 && (
+								<PaginationItem>
+									<PaginationEllipsis />
+								</PaginationItem>
+							)}
+							<PaginationItem className="hover:cursor-pointer">
+								<PaginationNext
+									onClick={() =>
+										currentPage < maxPage ? loadPage(currentPage + 1) : null
+									}
+								/>
+							</PaginationItem>
+						</PaginationContent>
+					</Pagination>
+				)}
+			</section>
+
+			<section>
+				{!getSelectedReports.length ? (
```

Comment:
why render the empty section if the array length is 0?
```suggestion
				{!getSelectedReports.length ? (
				<section>
```

Reply:
🙃
✅ updated